description/src/transformer.rs

```diff
@@ -12,7 +12,13 @@ pub struct Transformer<'a, R, S = ()> {
     cache: RefCell<HashMap<u32, Cached<R>>>,
     /// state can be used for example for an Rng
     pub state: S,
+    /// The `policy` defines, how to transform a type. If the type is unrepresentable, return an Err.
     policy: fn(u32, &Type<PortableForm>, &Self) -> anyhow::Result<R>,
+    /// The `recurse_policy` defines, how to handle cases,
+    /// where a type has been visited before, and is visited again BEFORE a representation of this type could be computed.
+    /// If it returns Some(..) we avoid infinite recursion by returning a concrete value.
+    /// If it returns None, nothing is done in the case of detected recursion and the type falls through.
```

Comment:
What does `falls through` mean here? Is the recursion continued as the type is "unresolved"?

For which containers could this happen? The blow "this is dangerous" seems a bit scary, might be worth a `# Warning` section

Reply:
Ah yes, the name is not so good, perhaps `recursion_should_continue` is better. On these types we should just let the recursion continue instead of intersecting. That is the idea behind it. I think having this on the containers `Array`, `Sequence`, `Tuple` and `Compact` should be fine, because they cannot recurse infinitely without going through a named type (e.g. a `Variant` or `Composite`) that will be intersected when recursed on.